powerauth-java-http/src/main/java/io/getlime/security/powerauth/http/validator/ValueTypeValidator.java

```diff
@@ -97,7 +97,26 @@ public static boolean isValidSignatureType(String signatureType) {
      * @return True if signature candidate has correct format, false otherwise.
      */
     public static boolean isValidSignatureValue(String signature) {
-        return signature != null && signature.matches(signatureRegex);
+        if (signature != null) {
+            switch (signature.length()) {
+                case 8:
+                case 17:
+                case 26:
+                    // "2.0", "2.1", "3.0" signature version uses "DECIMAL" format
+                    return signature.matches(signatureRegex);
+                case 24:
+                case 44:
+                case 64:
+                    // "3.1" and later signatures uses "BASE64" format.
+                    // We don't need to validate an exact number of encoded bytes. This is due to fact,
+                    // that if input string length can only be 24, 44 or 64, then the encoded output length
```

Comment:
Shouldn't the `encoded` word be `decoded`? We are checking length of string after the `base64.decode()` call.

Reply:
It's only an assumption that several string lengths are in direct relationship to an expected number of encoded bytes (someone encoded that bytes before). We're going to validate that expected number of encoded bytes. If it's faster, then I would use regexpt to validate that string, instead of full decode. From this point of view, it doesn't matter that the function actually decodes that bytes to validate the lengths.